addons/mail/static/src/models/autocomplete_input_view/autocomplete_input_view.js

```diff
@@ -0,0 +1,232 @@
+            }
+            if (this.discussAsChatMobileAddItemHeaderOwner) {
+                this.discussAsChatMobileAddItemHeaderOwner.handleAddChatAutocompleteSource(req, res);
+            }
+            if (this.discussSidebarCategoryAsAddingItemInputViewOwner) {
+                return this.discussSidebarCategoryAsAddingItemInputViewOwner.onAddItemAutocompleteSource(req, res);
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                const value = _.escape(req.term);
+                this.messaging.models['Partner'].imSearch({
+                    callback: partners => {
+                        const suggestions = partners.map(partner => {
+                            return {
+                                id: partner.id,
+                                value: partner.nameOrDisplayName,
+                                label: partner.nameOrDisplayName,
+                            };
+                        });
+                        res(_.sortBy(suggestions, 'label'));
+                    },
+                    keyword: value,
+                    limit: 10,
+                });
+            }
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computeCustomClass() {
+            if (this.discussSidebarCategoryAsAddingItemInputViewOwner) {
+                if (this.discussSidebarCategoryAsAddingItemInputViewOwner === this.messaging.discuss.categoryChannel) {
+                    return 'o_DiscussSidebarCategory_newChannelAutocompleteSuggestions';
+                }
+            }
+            if (this.messagingMenuAsMobileNewMessageInput) {
+                return 'o_MessagingMenu_mobileNewMessageInputAutocomplete';
+            }
+        },
+        /**
+         * @private
+         * @returns {boolean|FieldCommand}
+         */
+        _computeIsHtml() {
+            if (this.discussSidebarCategoryAsAddingItemInputViewOwner) {
+                return this.discussSidebarCategoryAsAddingItemInputViewOwner === this.messaging.discuss.categoryChannel;
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {string|FieldCommand}
+         */
+        _computePlaceholder() {
+            if (this.chatWindowOwnerAsNewMessageFormInputOwner) {
+                return this.env._t("Search user...");
+            }
+            if (this.discussAsChannelMobileAddItemHeaderOwner) {
+                return this.discussAsChannelMobileAddItemHeaderOwner.addChannelInputPlaceholder;
+            }
+            if (this.discussAsChatMobileAddItemHeaderOwner) {
+                return this.discussAsChatMobileAddItemHeaderOwner.addChatInputPlaceholder;
+            }
+            if (this.discussSidebarCategoryAsAddingItemInputViewOwner) {
+                return this.discussSidebarCategoryAsAddingItemInputViewOwner.newItemPlaceholderText;
+            }
```

Comment:
why are those fields instead of just hardcoded string like the others?

Reply:
Done (btw `addXInput` were `undefined` because forgot to move them from component to model, just pasted `env._t` string immediately)